Sources/Units/Unit/Fraction.swift

```diff
@@ -0,0 +1,163 @@
+
+
+public struct Fraction: Hashable, Equatable, Sendable {
+    let numerator: Int
+    let denominator: Int
+
+    init(numerator: Int, denominator: Int) {
+        let gcd = Self.gcd(numerator, denominator)
+        self.numerator = numerator / gcd
+        self.denominator = denominator / gcd
+    }
+
+    var positive: Bool {
+        switch (numerator, denominator) {
+            // 0/0 is not positive in this logic
```

Comment:
Just as an aside, here's a fun math Wikipedia article on this particular fraction: https://en.wikipedia.org/wiki/Indeterminate_form#Indeterminate_form_0/0